cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -950,7 +950,19 @@ class QueueIamSchema(IamSchema):
 class LoginNodesIamSchema(BaseIamSchema):
     """Represent the IAM schema of LoginNodes."""
 
-    instance_profile = fields.Str(validate=validate.Regexp("^arn:.*:instance-profile/"))
+    instance_role = fields.Str(
+        metadata={"update_policy": UpdatePolicy.LOGIN_NODES_STOP}, validate=validate.Regexp("^arn:.*:role/")
+    )
+
+    additional_iam_policies = fields.Nested(
+        AdditionalIamPolicySchema,
+        many=True,
+        metadata={"update_policy": UpdatePolicy.LOGIN_NODES_STOP, "update_key": "Policy"},
+    )
+
+    instance_profile = fields.Str(
+        metadata={"update_policy": UpdatePolicy.LOGIN_NODES_STOP}, validate=validate.Regexp("^arn:.*:instance-profile/")
+    )
```

Comment:
Since these regex are used in many places, it would be better to define them in `constants.py` and use the same constant everywhere.